packages/atomic/src/utils/store.ts

```diff
@@ -32,6 +32,11 @@ export type AtomicStore = {
   facetElements: HTMLElement[];
   sortOptions: SortDropdownOption[];
   iconAssetsPath: string;
+  /**
+   * Flag set to 'true' when the first "atomic-result" & its component have been rendered, replacing the placeholders.
+   * When first loading, Atomic has a rendering delay of its results, hence this is a better indicator than the "firstSearchExecuted" of the Headless state.
+   */
+  firstResultLoaded: boolean;
```

Comment:
Instead of using an event, which I first implemented, I decided to use a store flag so it's more easily accessible by any Atomic component or the users themselves

Reply:
I realise that the "Store" isn't a property on the interface like the engine is, we could add this too.